packages/adaptive_scaffold/lib/src/slot_layout_config.dart

```diff
@@ -0,0 +1,69 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/widgets.dart';
+
+/// A Widget that takes a child and a Function that returns an animated Widget
+/// then displays the child with this animation.
+///
+/// Most commonly used in [LayoutSlot] however it would be functional outside of
+/// this Widget as well.
+class SlotLayoutConfig extends StatefulWidget {
+  /// Creates a new [SlotLayoutConfig].
+  ///
+  /// Returns the child widget as is but holds properties to be accessed by other
+  /// classes.
+  const SlotLayoutConfig({
+    required this.builder,
+    this.inAnimation,
+    this.outAnimation,
+    required super.key,
+  });
+
+  /// The child Widget that the parent eventually returns with an animation.
```

Comment:
Maybe explain that this child will coexist with the widget of another `SlotLayoutConfig` of the same slot during transition, especially if `GlobalKey` is involved.